main/boost/APKBUILD

```diff
@@ -46,27 +46,38 @@ _libs="
 	filesystem
 	graph
 	iostreams
+	locale
+	log_setup
+	log
 	math
 	prg_exec_monitor
 	program_options
 	python3
-	python
+	python2
 	random
 	regex
 	serialization
 	signals
+	stacktrace_basic
+	stacktrace_noop
 	system
 	thread
+	timer
+	type_erasure
 	unit_test_framework
 	wave
 	wserialization
 	"
 for _lib in $_libs; do
-	case "$_options_carch" in
-		*$_lib*) continue;;
-	esac
-	subpackages="$subpackages $pkgname-$_lib:_boostlib"
-done
+	_skip_lib=0
+	for _option in $_options_carch; do
+		case "$_option" in --without-${_lib}) _skip_lib=1; break ;; esac
+	done; unset -v _option
+	if [ ${_skip_lib:-0} -eq 0 ]; then
+		subpackages="$subpackages ${pkgname}-${_lib}:_boostlib"
+	fi
+done; unset -v _lib _skip_lib
+subpackages="$subpackages $pkgname-libs"
```

Comment:
I wonder what this change is supposed to solve in the first place? If you are afraid of overlapping matches, then why not do something like:
````sh
case "$_options_carch " in # the trailing space is intentionally there so we can match last word
    *"--without-$_lib "*) continue;;
esac
````

Reply:
I don't see any problem with using that kind of case instead.

Reply:
@ncopa Have a look at https://github.com/alpinelinux/aports/pull/4657/commits/658c6b24d67a901147dda84e9b967d494f09b606 and let me know which you prefer to use.

Reply:
The point of the change was to handle `_options_carch` containing new lines like we see in `_libs`. Preventing `--without-log` from matching for `--without-log_setup` is also important.